src/actions/tree.js

```diff
@@ -164,35 +166,56 @@ export const changeDateFilter = ({newMin = false, newMax = false, quickdraw = fa
       dateMaxNumeric: dates.dateMaxNumeric,
-      visibility: data.visibility,
-      visibilityVersion: data.visibilityVersion,
-      branchThickness: data.branchThickness,
-      branchThicknessVersion: data.branchThicknessVersion,
-      idxOfInViewRootNode: tree.idxOfInViewRootNode,
-      stateCountAttrs: Object.keys(controls.filters)
     };
-    if (controls.showTreeToo) {
-      const dataToo = calculateVisiblityAndBranchThickness(treeToo, controls, dates);
-      dispatchObj.tangleTipLookup = constructVisibleTipLookupBetweenTrees(tree.nodes, treeToo.nodes, data.visibility, dataToo.visibility);
-      dispatchObj.visibilityToo = dataToo.visibility;
-      dispatchObj.visibilityVersionToo = dataToo.visibilityVersion;
-      dispatchObj.branchThicknessToo = dataToo.branchThickness;
-      dispatchObj.branchThicknessVersionToo = dataToo.branchThicknessVersion;
+    dispatch(changeDateAction);
+
+    const changeVisibilityAndThickness = () => {
+      const data = calculateVisiblityAndBranchThickness(tree, controls, dates);
+      const changeVisibilityThicknessAction = {
+        type: types.CHANGE_DATES_VISIBILITY_THICKNESS,
+        quickdraw,
+        dateMin: newMin ? newMin : controls.dateMin,
+        dateMax: newMax ? newMax : controls.dateMax,
+        dateMinNumeric: dates.dateMinNumeric,
+        dateMaxNumeric: dates.dateMaxNumeric,
+        visibility: data.visibility,
+        visibilityVersion: data.visibilityVersion,
+        branchThickness: data.branchThickness,
+        branchThicknessVersion: data.branchThicknessVersion,
+        idxOfInViewRootNode: tree.idxOfInViewRootNode,
+        stateCountAttrs: Object.keys(controls.filters)
+      };
+      if (controls.showTreeToo) {
+        const dataToo = calculateVisiblityAndBranchThickness(treeToo, controls, dates);
+        changeVisibilityThicknessAction.tangleTipLookup =
+          constructVisibleTipLookupBetweenTrees(tree.nodes, treeToo.nodes, data.visibility, dataToo.visibility);
+        changeVisibilityThicknessAction.visibilityToo = dataToo.visibility;
+        changeVisibilityThicknessAction.visibilityVersionToo = dataToo.visibilityVersion;
+        changeVisibilityThicknessAction.branchThicknessToo = dataToo.branchThickness;
+        changeVisibilityThicknessAction.branchThicknessVersionToo = dataToo.branchThicknessVersion;
+      }
+      /* D I S P A T C H */
+      dispatch(changeVisibilityThicknessAction)
+      updateEntropyVisibility(dispatch, getState);
+      if (frequencies.loaded) {
+        updateFrequencyDataDebounced(dispatch, getState);
+      }
     }
 
-    /* D I S P A T C H */
-    dispatch(dispatchObj);
-    updateEntropyVisibility(dispatch, getState);
-    if (frequencies.loaded) {
-      updateFrequencyDataDebounced(dispatch, getState);
+    clearTimeout(timeout)
+    // Quickdraw also means debounce in this context, but better change it.
+    // TODO: Refactor to two seperate variables for quickdraw and debounce
```

Comment:
@avin-kavish Definitely agree with the TODO here, I think it would make this more readable. Could you add that refactor here please? Thanks!